src/lseq/ident.rs

```diff
@@ -3,6 +3,7 @@ use bitvec::vec::*;
 use rand::Rng;
 use serde::{Deserialize, Serialize};
 
+// TODO: This boundary is way too strict, especially for very deeply nested trees. the boundary should grow exponentially with the tree
```

Comment:
The papers don't use a variable boundary I'm not sure that this is desirable either. It's really only meant to give a slight nudge _on top_ of random allocation.

Reply:
Hmm, ok I think we should do a bit of our own analysis on these strategies after things have stabilized, some tuning might be nice to do